functorch/csrc/BatchRulesViews.cpp

```diff
@@ -435,14 +486,28 @@ std::tuple<Tensor, optional<int64_t>> view_batching_rule(
   return std::make_tuple(self_.view(size_), 0);
 }
 
+std::tuple<Tensor, optional<int64_t>> view_symint_batching_rule(
+    const Tensor& self,
+    optional<int64_t> self_bdim,
+    c10::SymIntArrayRef size) {
+  return self.view(self_bdim, c10::asIntArrayRefSlow(size));
+}
```

Comment:
You do want to do what is done in [this gist](https://gist.github.com/zou3519/75b446ec3627a783cebe76fc61aa2e54) because m.impl will require that the function passed to it (view_symint_batching_rule) has the same schema as the operator (view.SymInt) when it exists.

The VMAP_SUPPORT macro does some template magic that makes it so that the function passed has a different signature than the Schema

Reply:
Got it! Copy-pasted yr version!